src/Guide/Api/Methods.hs

```diff
@@ -166,9 +174,21 @@ createTrait db itemId traitType text = do
 -- | Update the text of a trait (pro/con).
 setTrait :: DB -> Uid Item -> Uid Trait -> Text -> Handler NoContent
 setTrait db itemId traitId text = do
-  (_edit, _newTrait) <- dbUpdate db (SetTraitContent itemId traitId text)
-  -- TODO diff and merge
-  pure NoContent
+  dbQuery db (GetItemMaybe itemId) >>= \case
+    Nothing -> throwError (err404 {errBody = "Item not found"})
+    Just _ -> do
+      dbQuery db (GetTraitMaybe itemId traitId) >>= \case
+        Nothing -> throwError (err404 {errBody = "Trait not found"})
+        Just original -> do
+          let fromMarkDown = markdownInlineMdSource . _traitContent
+          modified <- dbQuery db (GetTrait itemId traitId)
```

Comment:
`modified` and `original` are going to be the same here.

The server doesn't know what the original trait was, because "original" = "what the user saw when they started editing", and the frontend doesn't notify the server when the user starts editing.

You have to expect that the user will send you both `original` and `modified`. You can create a client type for that: `CTextEdit { original :: Text, modified :: Text }`.

Reply:
Don't forget to update the other handler, too.